src/System.Text.Primitives/System/Text/FormattingData.cs

```diff
@@ -93,6 +286,59 @@ public static FormattingData InvariantUtf8
             }
         }
 
+        /// <summary>
+        /// Parse the next byte in a byte array. Will return either a DigitOrSymbol value, an InvalidCharacter, or a Continue
+        /// </summary>
+        /// <param name="nextByte">The next byte to be parsed</param>
+        /// <param name="bytesParsed">The total number of bytes parsed (will be zero until a code unit is deciphered)</param>
+        /// <returns></returns>
+        public uint ParseNextCodingUnit(ref byte[] buffer, ref int bufferIndex)
```

Comment:
>ParseNextCodingUnit [](start = 20, length = 19)

Please add a TODO to convert it to TryParse(byte[], int, out CodePoint, out int)